tests/scripts/thread-cert/node.py

```diff
@@ -175,6 +175,55 @@ def _expect(self, pattern, timeout=-1, *args, **kwargs):
                 if timeout <= 0:
                     raise
 
+    def _prepare_pattern(self, pattern):
+        """Build a new pexpect pattern matching line by line.
+
+        Adds lookahead and lookbehind to make each pattern match a whole line,
+        and add 'Done' as the first pattern.
+
+        Args:
+            pattern: a single regex or a list of regex.
+
+        Returns:
+            A list of regex.
+        """
+        EXPECT_LINE_FORMAT = r'(?<=[\r\n])%s(?=[\r\n])'
+
+        if isinstance(pattern, list):
+            pattern = [EXPECT_LINE_FORMAT % p for p in pattern]
+        else:
+            pattern = [EXPECT_LINE_FORMAT % pattern]
+
+        return [EXPECT_LINE_FORMAT % 'Done'] + pattern
```

Comment:
Should we also add `Error:.*` along side with `Done` ?

Reply:
I don't think so. Error is not expected on success, and the test should fail for unexpected results.

Reply:
> I don't think so. Error is not expected on success, and the test should fail for unexpected results.

Agree that we should fail for `Error: ...`. That was what I mean by adding `Error: ...`. Maybe we can enhance it later.